functions/helpFunctions.tsx

```diff
@@ -191,3 +192,38 @@ export function chooseCorrelatedAssessment(
   }
   return null;
 }
+
+export function uploadDataToLocalstorage() {
+  const totalTasks: number =
+    data.ext_inspera_candidates[0].result.ext_inspera_questions.length;
+  const taskNumbers: number[] = Array.from(Array(totalTasks).keys());
+
+  //check if data is already uploaded to localstorage
+  if (typeof window !== 'undefined') {
+    taskNumbers.map((taskNum: number) => {
+      const key: string = (taskNum + 1).toString() + '_data';
+      const dataToBeUploaded: AnswerType[] = [];
+      // data is not uploaded
+      if (!(key in localStorage)) {
+        data.ext_inspera_candidates.map((candidate: any) => {
+          // create answer objects and add to list
+          dataToBeUploaded.push({
+            assessmentId: uuidv4(),
+            answer: replaceUndefined(
```

Comment:
I do think `dataToBeUpdated` needs the `isFlagged` attribute (can initially be false) to be a propper `AnswerType`. Weird that the editor hasn't given a response to it when `dataToBeUpdated` is `AnswerType[]`!

Reply:
![image](https://user-images.githubusercontent.com/44196526/161738876-41f96664-4c88-487d-a087-9a0ed75be702.png)

`AnswerType` does not have that attribute, but `AssessmentType` does.